File: src/model/pieces/knight.py
from src.model.pieces.piece import Piece


class Knight(Piece):
    def __init__(self, color, em):
        key_letter = "n"
        super().__init__(color, key_letter, em)
        self.moves = [
            ([1, 2], False),
            ([2, 1], False),
            ([2, -1], False),
            ([-1, 2], False),
            ([1, -2], False),
            ([-1, -2], False),
            ([-2, -1], False),
            ([-2, 1], False)
        ]

File: src/pieces/knight.js
import Piece from '../piece.js'
import SingleMovement from '../movement/singleMovement.js'

export default class Knight extends Piece {
  constructor(color) {
    super(color, [
      new SingleMovement(2, 1),
      new SingleMovement(-2, 1),
      new SingleMovement(2, -1),
      new SingleMovement(-2, -1),
      new SingleMovement(1, 2),
      new SingleMovement(-1, 2),
      new SingleMovement(1, -2),
      new SingleMovement(-1, -2),
    ])
  }

  toString() {
    return this.color === 'black' ? '♘' : '♞'
  }
}
